src/libcmd/installable-attr-path.cc

```diff
@@ -48,6 +48,26 @@ DerivedPathsWithInfo InstallableAttrPath::toDerivedPaths()
 {
     auto v = toValue(*state).first;
 
+    if (v->type() == nPath) {
+        auto storePath = v->path().fetchToStore(state->store);
+        return {{
+            .path = DerivedPath::Opaque {
+                .path = std::move(storePath),
+            },
+            .info = make_ref<ExtraPathInfo>(),
+        }};
+    }
+
+    else if (v->type() == nString) {
+        return {{
+            .path = state->coerceToDerivedPath(
+                noPos,
+                *v,
+                ""),
+            .info = make_ref<ExtraPathInfo>(),
+        }};
+    }
```

Comment:
These two conditionals duplicate the equivalent flake attribute logic, except for the `errorCtx` string, which should remain a parameter.
`std::optional<DerivedPathsWithInfo> trySinglePathToDerivedPaths(Value *v, string_view errorCtx)`?